app/routes.py

```diff
@@ -1,2 +1,234 @@
+        return make_response({"details":"Give video id"}, 400)
+
+    if a_customer_id == None or not isinstance(a_customer_id, int):
+        return make_response({"details":"Give video id"}, 400)
+
+    a_video = Video.query.get_or_404(a_video_id)
+    a_customer = Customer.query.get_or_404(a_customer_id)
+        
+    rental = Rental.query.filter_by(customer_id = a_customer_id, video_id = a_video_id).one_or_none()
+    if not rental:
+        return make_response({"details": "This rental doesn't exist"}, 400)
+    checked_in_video = rental
+
+    a_video.available_inventory += 1
+    a_customer.videos_checked_out_count -= 1
+
+    db.session.delete(rental)
+    db.session.commit()
+
+    response = {
+        "customer_id": checked_in_video.customer_id,
+        "video_id": checked_in_video.video_id,
+        "videos_checked_out_count": a_customer.videos_checked_out_count,
+        "available_inventory": a_video.available_inventory
+    }
+
+    return jsonify(
+        response
+    ), 200
+
+@customer_bp.route("/<customer_id>/rentals", methods=["GET"], strict_slashes=False)
+def customer_rentals(customer_id):
+    if not Customer.query.get(customer_id):
+        return make_response({"details": "Customer does not exist"}, 404)
+
+    #use the join query for customer. pattern customer, video, customer
+    all_rentals = db.session.query(Rental).join(Customer, Customer.customer_id==Rental.customer_id)\
+        .join(Video, Video.video_id==Rental.video_id).filter(Customer.customer_id == customer_id).all()
+
+    list_of_rentals = []
+    for rental in all_rentals:
+        #create the return statement and to get the values the set up is the same how rental response is set up. but instead of self use all_rentals. because of the join clause
+        list_of_rentals.append({
+            "release_date": Video.query.get(rental.video_id).release_date,
+            "title": Video.query.get(rental.video_id).title,
+            "due_date": rental.due_date
+        })
+    return jsonify(list_of_rentals), 200
+
+@video_bp.route("/<video_id>/rentals", methods=["GET"], strict_slashes=False)
+def video_rentals(video_id):
+    if not Video.query.get(video_id):
+        return make_response({"deatils": "Video does not exist"}, 404)
+
+    #do the same as above just change the filter part so its for videos
+    all_rentals = db.session.query(Rental).join(Video, Video.video_id==Rental.video_id)\
+        .join(Customer, Customer.customer_id==Rental.customer_id).filter(Video.video_id == video_id).all()
+
+    list_of_rentals = []
+    for rental in all_rentals:
+        list_of_rentals.append({
+            "due_date": rental.due_date,
+            "name": Customer.query.get(rental.customer_id).name,
+            "phone": Customer.query.get(rental.customer_id).phone,
+            "postal_code": Customer.query.get(rental.customer_id).postal_code
+        })
+    return jsonify(list_of_rentals), 200
```

Comment:
Multiple `get()` requests aren't necessary here also.
```suggestion

    for rental in all_rentals:
        customer = Customer.query.get(rental.customer_id)
        list_of_rentals.append({
            "due_date": rental.due_date,
            "name": customer.name,
            "phone": customer.phone,
            "postal_code": customer.postal_code
        })
    return jsonify(list_of_rentals), 200
```